### DISCUSSION TITLE
Is there a method/function to output normalized adjacency matrix directly for sampling?

### DISCUSSION THREAD

**Discussion by RX28666** [pyg-team/pytorch_geometric#5954] [Q&A] [answered]
Hello, I try to get the symmetrically normalized adjacency with self loops: `D^(-1/2)(A+I)D^(-1/2)` for each batch after sampling, Neighbor loader outputs edge-index based on current batch, so I compute the normalized adjacency matrix using my own function for each batch, but it seems not to be very effective because it has to be computed for each batch, especially when batch size is large, I am wondering if there is another way to get normalized adjacency matrix for each batch. Thanks for help.

**Comment by wsad1** [comment, marked as answer]
Since each batch is a different sugbgraph, you'll have to compute the normalized adjacency each time.

**Reply by RX28666** [reply]
Thanks!
